### DISCUSSION TITLE
How to remove the internal color filter for SPHERICAL_MERCATOR profile?

### DISCUSSION THREAD

**Discussion by JoviLeejw** [pelicanmapping/osgearth#2335] [Support Q&A]
I added an XYZImageLayer with the following code snippet. I did not add any ColorFilter to this image layer. But it seems like there is an internal color filter that makes the color different from the original map tile. Is there a way to keep the original tile color?

- osgEarth: 3.4
- osg: 3.6.5(GL3 build)
- os: Mac OS 13.5.2
- XCode: 14.1 

```
// add a semi-transparent XYZ layer:
    XYZImageLayer* osm = new XYZImageLayer();
    osm->setURL("http://[abc].tile.openstreetmap.org/{z}/{x}/{y}.png");
    osm->setProfile(Profile::create(Profile::SPHERICAL_MERCATOR));
    osm->setOpacity(1.0f);
    map->addLayer(osm);
```
The effect is like this. 
<img width="477" alt="image" src="https://github.com/gwaldron/osgearth/assets/78395455/4d1d166a-d29a-440f-b5e4-3cc8bb9cfa4c">
The original map tile. 
https://a.tile.openstreetmap.org/0/0/0.png
<img width="256" alt="image" src="https://github.com/gwaldron/osgearth/assets/78395455/6f309b00-74c7-4a11-8d37-9f2d4559ffd6">

Same issue for the satellite image tile in SPHERICAL_MERCATOR profile
<img width="435" alt="image" src="https://github.com/gwaldron/osgearth/assets/78395455/845186bc-a7d5-409e-86e7-7174253422cb">
<img width="256" alt="image" src="https://github.com/gwaldron/osgearth/assets/78395455/2d151ca0-bd48-40b6-ae50-c0e7524939f2">

No such issue if I add a TMSImageLayer not in SPHERICAL_MERCATOR profile.
<img width="442" alt="image" src="https://github.com/gwaldron/osgearth/assets/78395455/cf6a2dc3-658b-4c5e-a3e5-ebe40673426d">

Hopefully receive your feedback, many thanks!

Jovi Lee

**Comment by JoviLeejw** [comment]
I finally found the root cause. No relationship with osgEarth. It's an imageio plugin issue on Mac OS.
It's really frustrating.
```
auto image = osgDB::readImageFile("/Users/jovi/Project/OSG/URL1.jpeg");
    if (image) {
        osgDB::writeImageFile(*image, "/Users/jvoi/Project/OSG/URL2.jpeg");
    }
```
<img width="701" alt="image" src="https://github.com/gwaldron/osgearth/assets/78395455/c0cc966f-5646-4658-9a8c-a48c7ae0e983">

**Comment by gwaldron** [comment]
Glad you found the cause of the issue, because we had no idea :)
That is really strange...

**Reply by JoviLeejw** [reply]
Hi @gwaldron, I fixed this issue by adding the following code snippet in osgEarth::GDAL::reprojectImage().
The previous reprojected image has a different pixel format(GL_RGBA) from the src image(GL_BGRA).
Do you think the change is reasonable?
```
    if (result && srcImage) {
        result->setPixelFormat(srcImage->getPixelFormat());
        result->setInternalTextureFormat(srcImage->getInternalTextureFormat());
    }
```
<img width="675" alt="image" src="https://github.com/gwaldron/osgearth/assets/78395455/b106ab63-5e69-43fa-a295-caba61c8a14e">
